distro/runtime/concierge/start.sh

```diff
@@ -1,51 +1,197 @@
+#   * when called with "debug" option as first argument:
+#     * will enable Java Flight Recorder (for Oracle JVM only if installed)
+#     * will enable JMX for remote monitoring (when running at least compact3 profile)
+#
+# Returned error codes:
+#   1: $JAVA_HOME/bin/java not found
+#   2: Java Version < 1.8
+#   3: Java Compact 2 at least needed
+#
+
 DIRNAME=`dirname "$0"`
 PROGNAME=`basename "$0"`
 
 RUNTIME_FOLDER=`cd "$DIRNAME/.."; pwd`
 BASE_FOLDER=`cd "$RUNTIME_FOLDER/.."; pwd`
 
-# set ports for HTTP(S) serverputt
+# set ports for HTTP(S) server
 if [ "x$HTTP_PORT" = "x" ]; then
     HTTP_PORT="8080"
 fi
-
 if [ "x$HTTPS_PORT" = "x" ]; then
     HTTPS_PORT="8443"
 fi
 
-# Check Java version. Make sure java command is available
-echo "Using java: $JAVA_HOME" && `java -version`
-VERSION=`java -version 2>&1  | egrep '"([0-9].[0-9]\..*[0-9]).*"' | awk '{print substr($3,2,length($3)-2)}' | awk '{print substr($1, 3, 3)}' | sed -e 's;\.;;g'`
-if [ "$VERSION" -lt "80" ]; then
-   echo "ERROR: JVM must be greater than 1.7"
-   exit 1;
-fi
+# arg 0: debug enabled: true/false
+findJvmVersion() {
+    # use java from JAVA_HOME if set, otherwise java from path, e.g. default java on RaspPi
+    JAVA_BIN=java
+    if [ "x$JAVA_HOME" != "x" ] ; then
+        JAVA_BIN=$JAVA_HOME/bin/java
+        if [ ! -f $JAVA_BIN ] ; then
+            echo "ERROR: could not find $JAVA_BIN"
+            exit 1
+        fi
+    fi
+
+    # Check Java version. Make sure java command is available
+    JAVA_VERSION_OUTPUT=`$JAVA_BIN -version 2>&1`
+    echo "Using java: $JAVA_BIN"
+    echo "$JAVA_VERSION_OUTPUT"
+
+    JAVA_VERSION=`echo $JAVA_VERSION_OUTPUT | egrep '"([0-9].[0-9]\..*[0-9]).*"' | awk '{print substr($3,2,length($3)-2)}' | awk '{print substr($1, 3, 3)}' | sed -e 's;\.;;g'`
+    # get information about compact profile (1/2/3/fulljre) and Vendor (Oracle, Azul)
+    JAVA_COMPACT_PROFILE="fulljre"
+    JAVA_VENDOR="Oracle"
+
+    IS_COMPACT1_PROFILE=`echo $JAVA_VERSION_OUTPUT | sed -e 's/^.*compact1.*$/true/g'`
+    if [ "$IS_COMPACT1_PROFILE" = "true" ] ; then JAVA_COMPACT_PROFILE="compact1" ; fi
+    IS_COMPACT2_PROFILE=`echo $JAVA_VERSION_OUTPUT | sed -e 's/^.*compact2.*$/true/g'`
+    if [ "$IS_COMPACT2_PROFILE" = "true" ] ; then JAVA_COMPACT_PROFILE="compact2" ; fi
+    IS_COMPACT3_PROFILE=`echo $JAVA_VERSION_OUTPUT | sed -e 's/^.*compact3.*$/true/g'`
+    if [ "$IS_COMPACT3_PROFILE" = "true" ] ; then JAVA_COMPACT_PROFILE="compact3" ; fi
+    echo "JAVA_COMPACT_PROFILE=$JAVA_COMPACT_PROFILE"
+
+    IS_AZUL_JAVASE=`echo $JAVA_VERSION_OUTPUT | sed -e 's/^.*Zulu-Embedded.*$/true/g'`
+    if [ "$IS_AZUL_JAVASE" = "true" ] ; then JAVA_VENDOR="Azul" ; fi
+    echo "JAVA_VENDOR=$JAVA_VENDOR"
+}
+
+checkJvmVersion() {
+    # Minimal requirement: Java8 with compact2 profile
+    if [ "$JAVA_VERSION" -lt "80" ]; then
+        echo "ERROR: JVM must be greater than 1.7"
+        exit 2;
+    fi
+    if [ "$JAVA_COMPACT_PROFILE" = "compact1" ]; then
```

Comment:
As pointed out in previous comments: up to now due to logback and Jersey a Full-JRE is required.
So we have to check here for fulljre until this has been solved
`if [ "$JAVA_COMPACT_PROFILE" != "fulljre" ]; then`